src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/unsubscribe_button_request.cc

```diff
@@ -93,35 +94,60 @@ void UnsubscribeButtonRequest::Run() {
     return;
   }
 
-  if (!app->UnsubscribeFromButton(
-          static_cast<mobile_apis::ButtonName::eType>(btn_id))) {
-    SDL_LOG_ERROR("App doesn't subscribe to button " << btn_id);
+  if (!app->IsSubscribedToButton(btn_id)) {
+    SDL_LOG_ERROR("App is not subscribed to button " << btn_id);
     SendResponse(false, mobile_apis::Result::IGNORED);
     return;
   }
```

Comment:
Since CUSTOM_BUTTONs must now be in the capabilities of an HMI, unsubscribing from them is possible. I propose adding a check such as this:

```
if (mobile_apis::ButtonName::CUSTOM_BUTTON == btn_id) {
    SDL_LOG_ERROR("App cannot unsubscribe from CUSTOM_BUTTON");
    SendResponse(false, mobile_apis::Result::IGNORED);
    return;
  }
```

Reply:
@iCollin from my point of view, HMI is now responsible for button subscriptions (including custom button).
So we concluded that SDL should have more generic logic and forward subscription requests for all buttons in the same way. For that reason we allowed applications to unsubscribe from CUSTOM_BUTTON. Additionally, if HMI rejects subscription to custom button during app registration - it will also be not subscribed to a custom button, so this case is possible anyway.
I propose to keep more generic logic on SDL side and (if needed) add this check on HMI side.

Reply:
Good idea, I think that makes sense as well.

Reply:
@AKalinich-Luxoft Hello, my team does not agree this is an acceptable solution.

Their position is that leaving this responsibility to HMI developers will lead to issues when it is forgotten or missed. Currently, SDL Core does not allow applications to unsubscribe from `CUSTOM_BUTTON` and nothing in this proposal suggests that behavior should change.

Can you please add a block like I suggested above to this PR?

Reply:
A check that will just prevent the `app->UnsubscribeButton` effect would suffice as well, the result code does not need to be modified.

Reply:
@iCollin this check has been added in https://github.com/smartdevicelink/sdl_core/pull/3743/commits/ce9fc4455093d4914dfec55a6e5003da3c28fc83